css/how_to_name_the_classes.md

```diff
@@ -0,0 +1,118 @@
+# How to name the classes?
+
+To understand which is the best name for a css class
+
+## 1. First rule: don't use id's, use classes
+
+The CSS language was born to avoid the repetition of inline style attributes where possible, using an id based css means that your code will be limited to a single part of your project, and since each id can be inserted in the page just once, that css will be used only one time.
+
+Creating your code using classes makes your project modular (or modular-ready), it will be easier for you and other developers to export it into multiple parts of your project or even into multiple projects
+
+## 2. The right name
+
+To choose the right name for a class we need to take into consideration the following suggestions
+
+### A. Allowed characters
+
+To avoid undesired errors, use only lowercase Unicode characters for the words and a single `-` hyphen character between each of them, in order to improve the language accessibility, we recommend to use english words.
+
+| Don't use | Acceptable | Recommended |
+| --------- | ---------- | ----------- |
+| niño      | nino       | kid         |
+| schön     | schon      | beautiful   |
+| délicieux | delicieux  | delicious   |
+
+### B. Prefix conflicts
+
+To prevent conflicts with other projects, define a prefix with the name of the module you are working on, and use it for every class that belong to that context. If the name is too long, use an abbreviation of it. If the prefix is too common, combine it with the project's name.
+
+```scss
+// SCSS example working on Fjord theme
+.fjord-button {...}
+.fjord-image {...}
+.fjord-text {...}
+```
+
+### C. Suffix modifiers
+
+Keep the name as simple as possible and add suffixes to create modifiers of the class
+
+```scss
+// Class base SCSS
+.fjord-btn {...} // stands for: .fjord-button
+```
+
+The general rule is `context-component-modifier`
+
+```scss
+// Class modifiers SCSS
+.fjord-btn-sm {...} // stands for: .fjord-button-small
+.fjord-btn-lg {...} // stands for: .fjord-button-large
+```
+
+If the variations of the class are too many, use a numeric suffix
+
+```scss
+// Class numeric modifiers SCSS
+.fjord-text-1 {...}
+.fjord-text-2 {...}
+.fjord-text-3 {...}
+.fjord-text-4 {...}
+```
+
+### D. Component instances
+
+We always recommend the solution proposed in the previous section, but sometimes it will be necessary to change the standard method
+
+> Imagine we need to create 2 images in 2 different components for the Fjord theme, but the differences between these elements are so uniques that we can't afford to generate the required modifiers because they won't be used anywhere else.
+
+In that case we are going to rely on the Atomic Design concepts to add the desired specifications to the class.
```

Comment:
Would be a good place to make the words "Atomic Design" a link to a resource on Atomic Design.